File: ada-ui/src/app/classes/class-metric-value.ts
export class ClassMetricValue {
  constructor(classDeclarationElementJSON: JSON) {
    this.numberOfPackageImportsIncoming = classDeclarationElementJSON['numberOfPackageImportsIncoming'];
    this.numberOfPackageImportsOutgoing = classDeclarationElementJSON['numberOfPackageImportsOutgoing'];

    this.numberOfAttributeInvocationsIncoming = classDeclarationElementJSON['numberOfAttributeInvocationsIncoming'];
    this.numberOfAttributeInvocationsOutgoing = classDeclarationElementJSON['numberOfAttributeInvocationsOutgoing'];

    this.numberOfMethodInvocationsIncoming = classDeclarationElementJSON['numberOfMethodInvocationsIncoming'];
    this.numberOfMethodInvocationsOutgoing = classDeclarationElementJSON['numberOfMethodInvocationsOutgoing'];

    this.numberOfConstructorInvocationsIncoming = classDeclarationElementJSON['numberOfConstructorInvocationsIncoming'];
    this.numberOfConstructorInvocationsOutgoing = classDeclarationElementJSON['numberOfConstructorInvocationsOutgoing'];

    this.bidirectionalNumberOfPackageImports = classDeclarationElementJSON['bidirectionalNumberOfPackageImports'];

    this.bidirectionalNumberOfAttributeInvocations = classDeclarationElementJSON['bidirectionalNumberOfAttributeInvocations'];

    this.bidirectionalNumberOfMethodInvocations = classDeclarationElementJSON['bidirectionalNumberOfMethodInvocations'];

    this.bidirectionalNumberOfConstructorInvocations = classDeclarationElementJSON['bidirectionalNumberOfConstructorInvocations'];

  }

  numberOfPackageImportsIncoming: number;
  numberOfPackageImportsOutgoing: number;

  numberOfAttributeInvocationsIncoming: number;
  numberOfAttributeInvocationsOutgoing: number;

  numberOfMethodInvocationsIncoming: number;
  numberOfMethodInvocationsOutgoing: number;

  numberOfConstructorInvocationsIncoming: number;
  numberOfConstructorInvocationsOutgoing: number;

  bidirectionalNumberOfPackageImports: number;

  bidirectionalNumberOfAttributeInvocations: number;

  bidirectionalNumberOfMethodInvocations: number;

  bidirectionalNumberOfConstructorInvocations: number;
}

File: ada-model/src/main/java/com/ucl/ADA/model/metrics/class_metrics/ClassMetricValue.java
package com.ucl.ADA.model.metrics.class_metrics;

import com.ucl.ADA.model.base_entity.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "CLASS_METRIC_VALUE")
public class ClassMetricValue extends BaseEntity {

    /**
     * total number of incoming package imports
     */
    @Column(name = "number_of_package_imports_incoming", nullable = false)
    private Float numberOfPackageImportsIncoming = 0f;

    /**
     * total number of outgoing package imports
     */
    @Column(name = "number_of_package_imports_outgoing", nullable = false)
    private Float numberOfPackageImportsOutgoing = 0f;

    /**
     * total number of incoming attribute invocations
     */
    @Column(name = "number_of_attribute_invocation_incoming", nullable = false)
    private Float numberOfAttributeInvocationsIncoming = 0f;

    /**
     * total number of outgoing attribute invocations
     */
    @Column(name = "number_of_attribute_invocation_outgoing", nullable = false)
    private Float numberOfAttributeInvocationsOutgoing = 0f;


    /**
     * total number of incoming method invocations
     */
    @Column(name = "number_of_method_invocation_incoming", nullable = false)
    private Float numberOfMethodInvocationsIncoming = 0f;

    /**
     * total number of outgoing method invocations
     */
    @Column(name = "number_of_method_invocation_outgoing", nullable = false)
    private Float numberOfMethodInvocationsOutgoing = 0f;


    /**
     * total number of incoming constructor imports
     */
    @Column(name = "number_of_constructor_invocation_incoming", nullable = false)
    private Float numberOfConstructorInvocationsIncoming = 0f;

    /**
     * total number of outgoing constructor invocations
     */
    @Column(name = "number_of_constructor_invocation_outgoing", nullable = false)
    private Float numberOfConstructorInvocationsOutgoing = 0f;


    /**
     * sum of numbers of incoming and outgoing package imports
     */
    @Column(name = "bidirectional_number_of_package_imports", nullable = false)
    private Float bidirectionalNumberOfPackageImports = 0f;

    /**
     * sum of numbers of incoming and outgoing attribute invocations
     */
    @Column(name = "bidirectional_number_of_attribute_invocations", nullable = false)
    private Float bidirectionalNumberOfAttributeInvocations = 0f;

    /**
     * sum of numbers of incoming and outgoing method invocations
     */
    @Column(name = "bidirectional_number_of_method_invocations", nullable = false)
    private Float bidirectionalNumberOfMethodInvocations = 0f;

    /**
     * sum of numbers of incoming and outgoing constructor invocations
     */
    @Column(name = "bidirectional_number_of_constructor_invocations", nullable = false)
    private Float bidirectionalNumberOfConstructorInvocations = 0f;

}
